test/unit/preprocessing/test_preprocessing_fairness.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable, Union
+
+import numpy as np
+import pandas as pd
+import pytest
+from numpy.typing import NDArray
+from scipy.stats import wilcoxon
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from typing_extensions import TypeAlias
+
+from fairlearn.datasets import fetch_adult
+from fairlearn.metrics import (
+    demographic_parity_difference,
+    equal_opportunity_difference,
+    equalized_odds_difference,
+)
+from fairlearn.preprocessing import CorrelationRemover, PrototypeRepresentationLearner
+
+PreprocessingAlgorithm: TypeAlias = Union[CorrelationRemover, PrototypeRepresentationLearner]
+
+
+@dataclass(frozen=True)
+class Dataset:
+    sensitive_feature: pd.Series
+    X_train: pd.DataFrame
+    X_test: pd.DataFrame
+    y_train: pd.Series
+    y_test: pd.Series
+
+
+@dataclass(frozen=True)
+class FairnessComparisonData:
+    sensitive_feature: pd.Series
+    y_true: pd.Series
+    y_pred_with_mitigation: NDArray
+    y_pred_without_mitigation: NDArray
+
+    def bootstrap(self, seed: int) -> FairnessComparisonData:
+        sample_indices = np.random.RandomState(seed=seed).choice(
```

Comment:
I remember recently reading that `numpy.random.default_rng` is the new way to do what `RandomState` did as they will support it further unlike the later. Won't change much for now, but maybe will make the future easier compatibility-wise?